src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcConnection.java

```diff
@@ -68,6 +70,16 @@ abstract class AbstractJdbcConnection extends AbstractJdbcWrapper
     this.usesDirectExecutor = ConnectionOptionsHelper.usesDirectExecutor(options);
   }
 
+  @Override
+  public DatabaseId getDatabaseId() {
+    return this.options.getDatabaseId();
+  }
+
+  @Override
+  public DatabaseClient getDatabaseClient() {
+    return getSpannerConnection().getDatabaseClient();
+  }
```

Comment:
interested to understand if there is a customer usecase where they would like to use databaseclient directly instead of using the connection.

Reply:
It is more of an 'escape hatch'. We try to support most features directly in the JDBC driver, but some Spanner features don't have a good equivalent in JDBC. In those cases, it can be easier to just use the database client directly. One such example could for example be `BatchWrite`. Although we can also create a method in the JDBC Connection that just delegates the call directly to the underlying database client, giving the option of just unwrapping the client ensures that all features are usable, even if there is no such delegate method.